src/repositories/versionable/VersionableRepository.ts

```diff
@@ -11,55 +11,33 @@ export default class VersionableRepository<D extends mongoose.Document, M extend
   count(): mongoose.Query<Number> {
     return this.modelType.countDocuments();
   };
-  findOne(query): mongoose.Query<D> {
-    return this.modelType.findOne(query);
+  async findOne(id: string) {
```

Comment:
cannot find by email like this in version repository. Versioning repository is common to all. cannot update it.